java/memory/memory-core/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -929,7 +957,7 @@ public ArrowBuf allocateBuffer() {
 
     @Override
     public int getSize() {
-      return nBytes;
+      return (int) nBytes;
```

Comment:
Do we have a safe cast utility? I'd rather use that than a blind cast

Reply:
We should. We have I think